minio/api.py

```diff
@@ -298,8 +298,11 @@ def list_buckets(self):
 
         if response.status != 200:
             raise ResponseError(response, method).get_exception()
-
-        return parse_list_buckets(response.data)
+        try:
+            return parse_list_buckets(response.data)
+        except InvalidXMLError:
+            if self._endpoint_url.endswith("s3.amazonaws.com"):
```

Comment:
We would need to check `access_key` and `secret_key` are `None` along with `endpoint_url`